net/src/parse.rs

```diff
@@ -59,6 +61,36 @@ pub(crate) trait ParsePayloadWith {
     fn parse_payload_with(&self, param: &Self::Param, cursor: &mut Reader) -> Option<Self::Next>;
 }
 
+pub(crate) trait ParseTarget {
+    fn call_parser<T>(cursor: &mut Reader) -> Option<Self>
```

Comment:
I need to think about this API a bit.

I think that it makes more sense to `impl<R> ParseTarget for R where R: Reader`

Reply:
and I would rename `call_parser` to `try_parse`